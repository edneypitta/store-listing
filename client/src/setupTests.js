import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

const RealDate = Date
export const mockDate = isoDate =>
  global.Date = class extends RealDate {
    constructor() {
      return new RealDate(isoDate)
    }
  }