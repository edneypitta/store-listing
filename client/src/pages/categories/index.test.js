import React from 'react'
import { render } from 'enzyme'
import CategoriesPage from './index'
import CategoriesContext from '../../categories-context'
import { mockDate } from '../../setupTests'
import { StaticRouter } from 'react-router-dom'

describe('CategoriesPage', () => {
  beforeEach(() => {
    mockDate('2019-03-05T12:00:00.977Z')
  })

  it('renders without crashing', () => {
    const category = { stores: [] }
    const component = render(
      categoriesPageComponent(category))

    expect(component).toMatchSnapshot()
  })
})

function categoriesPageComponent(category) {
  return (
    <CategoriesContext.Provider value={[category]}>
      <StaticRouter>
        <CategoriesPage />
      </StaticRouter>
    </CategoriesContext.Provider>)
}

