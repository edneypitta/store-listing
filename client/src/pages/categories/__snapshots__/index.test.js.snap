// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesPage renders without crashing 1`] = `
<div
  class="category-list"
>
  <div
    class="category"
  >
    <a
      class="link"
      href="/stores?category=undefined"
    >
      <div
        class="icon"
        title="Click to see stores"
      >
        <img
          alt="category icon"
        />
      </div>
      <div
        class="label"
      />
    </a>
  </div>
</div>
`;
